### PR TITLE
Models separation

### PR DESCRIPTION
**Pull request by Naerriel**

### CODE REVIEW

File: src/Models.js [hunk 1]
```diff
@@ -1,21 +1,54 @@
-function getExp(skillName){
-  /* Gets exp of a skill from storage.
-   */
+class skill{
```
Comment (andrzejgorski) [inline src/Models.js]: 1. Create new module global.js and place there only one variable:
```
const LifeGamification = {};
```
If you want to know why read this: 
https://stackoverflow.com/questions/5063878/how-to-cleanly-deal-with-global-variables
LifeGamification will be your namespaces for globals.

2. Replace all the instances of global variables from
`skillsFullInfo.push(smth);` to `LifeGamification.skillsFullInfo.push(smth);`

3. Wrap the all modules files by anonymous function as follow:
```
(function(){
  // Place the content of the module here. 
})();
```

and then
4. Create 
```class SkillsCollection{
  constructor(skills_array) {
    this.skills_array = skills_array;
  }

  new_skill(name) {
  // Creates new skill to collection
  // TODO implement it fully. 
  }
  save() {
  // Save Collection to the repository
  // It should looks like this
  // TODO implement it fully.
  LifeGamification.repository.save_skills_collection(this);
  }
  get_skill_by_id(id) {
  // This is comment for you not docstring. The name of the function explain everything. 
  // Iterate over all skills which will be placed in the array and return the one with id equal to function argument.
  // Forget about complexity of the algorithm it is not the OI.
  // TODO implement it fully.
  }
  get_skill_by_name(name) {
  // As in the previous funciton.
  // TODO implement it fully.
  }
```
Implement all functions of this class. You can add more function if necessary. 
5. Rename `skill` to `Skill`. The name of the classes should start with Capital letters.
6. In skillRename `nr` to `id`. This name is more meaningfull.
7. Add to the class Skill methods like add_exp() which updates current exp and level.
8. Add at the bottom of the module something like this:
```
LifeGamification.models = {
  skills_collection: new SkillsCollection(LifeGamification.repository.get_skills())
};
```
Ask me if you have any problems.

File: src/Models.js [hunk 2]
```diff
@@ -1,21 +1,54 @@
...
+      .then(function () {
+        if(i + 1 === skillsNames.length){
+          resolve();
+        }
+      });
+    }
+  });
+}
+
+function getLevelAndExpTillNextLevel(exp) {
```
Comment (andrzejgorski) [inline src/Models.js]: It should be a class method.

File: src/Models.js [hunk 3]
```diff
@@ -25,77 +58,49 @@ function getLevelAndLevelUpExp(exp) {
...
 
-function addSkill () {
-  /* Adds skill to storage.
+function updateExp(skillNr){
+  /* Increases skill's exp by the amount in correspondent text area.
    */
   return new Promise((resolve, reject) => {
-    let skillName = $('#skill_name').val();
-    $('#skill_name').val('');
+    let addedExp = parseInt($("#add_value_num" + skillNr).val());
```
Comment (andrzejgorski) [inline src/Models.js]: In models you can't use the functions from views like `$()`.

File: src/Models.js [hunk 4]
```diff
@@ -25,77 +58,49 @@ function getLevelAndLevelUpExp(exp) {
...
-}
-
-function updateSkill(skillNr){
-  /* Increases skill's exp by the amount in correspondent text area.
-   */
-  return new Promise((resolve, reject) => {
-    let addedExp = parseInt($("#add_value_num" + skillNr).val());
-    $("add_value_num" + skillNr).val('');
-    let skillName = allSkills[skillNr];
+    let skillName = $('#skill_name').val();
```
Comment (andrzejgorski) [inline src/Models.js]: Move skillName to the function argument.

File: src/Repository.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
+function getExp(skillName){
```
Comment (andrzejgorski) [inline src/Repository.js]: 1. Like in the models wrap it by code: 
```
(function (){

})();
```
2. Extend all function you need by:
```
  LifeGamification.repository = {};
  LifeGamification.repository.save_skills_collection = function(){

  };
```

File: src/popup.js [hunk 6]
```diff
@@ -1,7 +1,8 @@
 "use strict"
-var skillsArrayId = "skillsArrayId";
-var allSkills = [];
-var expTable = [];
+let skillsArrayId = "skillsArrayId";
+let skillsFullInfo = [];
+let skillsNames = [];
```
[the remark this answers is not in the document]
Comment (Naerriel) [reply src/popup.js]: ExpTable is needed. It contains information about how much exp is needed to reach certain level and this information isn't unique to each skill.
Do you have any suggestions on how to implement this level system without expTable?

**Pull request closed without merging**